tracepoint_decode/src/perf_abi.rs

```diff
@@ -0,0 +1,1409 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+// Adapted from linux/uapi/linux/perf_event.h.
+
+#![allow(non_upper_case_globals)]
+
+use core::fmt;
+use core::mem; // size_of
+
+use crate::PerfByteReader;
+
+/// perf_type_id: uint32 value for [`PerfEventAttr::attr_type`].
+#[repr(C)]
+#[derive(Clone, Copy, Debug, Default, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct PerfEventAttrType(pub u32);
+
+impl PerfEventAttrType {
+    /// PERF_TYPE_HARDWARE
+    pub const Hardware: Self = Self(0);
+
+    /// PERF_TYPE_SOFTWARE
+    pub const Software: Self = Self(1);
+
+    /// PERF_TYPE_TRACEPOINT
+    pub const Tracepoint: Self = Self(2);
+
+    /// PERF_TYPE_HW_CACHE
+    pub const HwCache: Self = Self(3);
+
+    /// PERF_TYPE_RAW
+    pub const Raw: Self = Self(4);
+
+    /// PERF_TYPE_BREAKPOINT
+    pub const Breakpoint: Self = Self(5);
+
+    /// Returns a string like "Hardware" or "Software" for the type.
```

Comment:
This feels like a ChatGPT comment.

Reply:
It wasn't, but maybe it should have been.